File: Clients/BlogBlazor/Shared/Blazor.Contracts/V1/Responses/UsersResponses/ApplicationUserResponse.cs
namespace Blazor.Contracts.V1.Responses.UsersResponses;

using System.Collections.Generic;
using Microsoft.AspNetCore.Identity;

/// <summary>
/// Application user response.
/// </summary>
public class ApplicationUserResponse
{
    /// <summary>
    /// Gets or sets first name.
    /// </summary>
    public string FirstName { get; set; }

    /// <summary>
    /// Gets or sets last name.
    /// </summary>
    public string LastName { get; set; }

    /// <summary>
    /// Gets or sets the email.
    /// </summary>
    /// <value>
    /// The email.
    /// </value>
    public string Email { get; set; }

    /// <summary>
    /// Gets or sets a value indicating whether [email confirmed].
    /// </summary>
    /// <value>
    ///   <c>true</c> if [email confirmed]; otherwise, <c>false</c>.
    /// </value>
    public bool EmailConfirmed { get; set; }

    /// <summary>
    /// Gets or sets roles.
    /// </summary>
    public ICollection<IdentityUserRole<string>> Roles { get; set; }

    /// <summary>
    /// Gets or sets the phone number.
    /// </summary>
    /// <value>
    /// The phone number.
    /// </value>
    public string PhoneNumber { get; set; }

    /// <summary>
    /// Gets or sets a value indicating whether [phone number confirmed].
    /// </summary>
    /// <value>
    ///   <c>true</c> if [phone number confirmed]; otherwise, <c>false</c>.
    /// </value>
    public bool PhoneNumberConfirmed { get; set; }

    /// <summary>
    /// Gets or sets the profile.
    /// </summary>
    /// <value>
    /// The profile.
    /// </value>
    public ProfileResponse Profile { get; set; }
}

File: Shared/Blog.FSharp.Contracts/V1/Responses/UsersResponses/ApplicationUserResponse.fs
namespace Blog.FSharp.Contracts.V1.Responses.UsersResponses

open System
open System.Collections.Generic
open Microsoft.AspNetCore.Identity

/// <summary>
/// Profile response.
/// </summary>
type ProfileResponse() =
    /// <summary>
    /// Gets or sets the user identifier.
    /// </summary>
    member val UserId: string = String.Empty with get, set

    /// <summary>
    /// Gets or sets the user.
    /// </summary>
    member val User: ApplicationUserResponse option = None with get, set

    /// <summary>
    /// Gets or sets the About.
    /// </summary>
    member val About: string = String.Empty with get, set

    /// <summary>
    /// Gets or sets the profile img.
    /// </summary>
    member val ProfileImg: string = String.Empty with get, set

/// <summary>
/// Application user response.
/// </summary>
and ApplicationUserResponse() =
    /// <summary>
    /// Gets or sets first name.
    /// </summary>
    member val FirstName: string = String.Empty with get, set

    /// <summary>
    /// Gets or sets last name.
    /// </summary>
    member val LastName: string = String.Empty with get, set

    /// <summary>
    /// Gets or sets the email.
    /// </summary>
    member val Email: string = String.Empty with get, set

    /// <summary>
    /// Gets or sets a value indicating whether email confirmed.
    /// </summary>
    member val EmailConfirmed: bool = false with get, set

    /// <summary>
    /// Gets or sets roles.
    /// </summary>
    member val Roles: ICollection<IdentityUserRole<string>> = List<IdentityUserRole<string>>() :> ICollection<_> with get, set

    /// <summary>
    /// Gets or sets the phone number.
    /// </summary>
    member val PhoneNumber: string = String.Empty with get, set

    /// <summary>
    /// Gets or sets a value indicating whether phone number confirmed.
    /// </summary>
    member val PhoneNumberConfirmed: bool = false with get, set

    /// <summary>
    /// Gets or sets the profile.
    /// </summary>
    member val Profile: ProfileResponse option = None with get, set